flink-yarn/src/test/java/org/apache/flink/yarn/YarnClusterDescriptorTest.java

```diff
@@ -511,6 +496,20 @@ public void testEnvironmentEmptyPluginsShipping() {
 		}
 	}
 
+	@Test
+	public void testDisableSystemClassPathIncludeUserJarAndWithIllegalShipDirectoryName() throws IOException {
+		final Configuration configuration = new Configuration();
+		configuration.setString(CLASSPATH_INCLUDE_USER_JAR, YarnConfigOptions.UserJarInclusion.DISABLED.toString());
+
+		final YarnClusterDescriptor yarnClusterDescriptor = createYarnClusterDescriptor(configuration);
+		try {
+			yarnClusterDescriptor.addShipFiles(Collections.singletonList(temporaryFolder.newFolder(ConfigConstants.DEFAULT_FLINK_USR_LIB_DIR)));
+			fail();
+		} catch (IllegalArgumentException exception) {
+			assertTrue(ExceptionUtils.findThrowableWithMessage(exception, "This is an illegal ship directory :").isPresent());
```

Comment:
I will simplify this to `assertThat(exception.getMessage(), containsString("This is an illegal ship directory :"));` before merging.